project/MimaExcludes.scala

```diff
@@ -34,6 +34,10 @@ import com.typesafe.tools.mima.core.ProblemFilters._
  */
 object MimaExcludes {
 
+  // Exclude rules for 2.4.x
```

Comment:
This reminds me that we should probably bump `previousSparkVersion` in `MimaBuild.scala` to be 2.2.0: https://github.com/apache/spark/blame/f340b6b3066033d40b7e163fd5fb68e9820adfb1/project/MimaBuild.scala#L91. I think this should happen for both master and branch-2.3.

See #15061 for an example of a similar change when 2.1.x was being prepared (it looks like we missed this step for 2.2.0). We may also need to un-exclude any new subprojects / artifacts that were added in 2.2.0 since they now need to be backwards-compatibility-tested for 2.3.x.

Reply:
We might want to make my proposed MiMa changes in a separate patch so the same set of reviewed changes can go to both master and branch-2.3.